sdks/java/container/agent/build.gradle

```diff
@@ -55,6 +39,22 @@ if (project.hasProperty('java11Home')) {
     }
 }
```

Comment:
I'm seeing CI failures for Java 17, e.g.

```
21:52:00 A problem occurred evaluating project ':sdks:java:container:agent'.
21:52:00 > Failed to apply plugin 'org.gradle.java'.
21:52:00    > Could not create task ':sdks:java:container:agent:compileJava'.
21:52:00       > Could not get unknown property 'errorprone' for task ':sdks:java:container:agent:compileJava' property 'options' of type org.gradle.api.tasks.compile.CompileOptions.
```

https://ci-beam.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Java17_Commit/4060/console

I think we need to _just_ set `javaVersion` before `applyJavaNature`, and continue to do the rest of the configuration after.

Reply:
Your right, its as if `javaVersion` should be a parameter to `applyJavaNature`